lam4-backend/test/PrinterSpec.hs

```diff
@@ -33,25 +33,22 @@ goldenGeneric name output_ = Golden
 
 spec :: Spec
 spec = do
-  test "predicate, no givens" predNoGivens "pred-no-givens"
-
-  where
-    test :: String -> String -> String -> Spec
-    test desc declStr fname = do
-      let decls :: [Decl] = parseCSTString declStr
-          printedDecls = printTree <$> decls
-      it desc $ goldenGeneric fname printedDecls
-
-
-predNoGivens :: String
-predNoGivens = [r|
-  DECIDE `pred blah blah`
-  // use the backtick syntax when you want an identifier with spaces
-  IF True|]
-
-
-parseCSTString :: String -> [Decl]
-parseCSTString str =
-  case evalParserFromScratch . parseProgramByteStr . BL.fromStrict . BS.pack $ str of
+  files <- runIO $ listDirectory "../examples"
```

Comment:
This tests automatically all the files in the example directory:
```
 Testing list2 [✔]
    Golden and Actual output didn't change
  Testing simple_join_or_record_field_deref [✔]
    Golden and Actual output didn't change
   …
  Testing chained_record_deref [✔]
    Golden and Actual output didn't change
```

(But all that it prints out is just empty strings for now, will fix that later.)